### ISSUE TITLE
Overlapping hits

### ISSUE THREAD

**Issue by ldenti** [knights-lab/BURST#35]
Hi,
I'm testing burst ALLPATHS mode but I'm getting less hits than the expected number.

These are the example data I'm using: [example.tar.gz](https://github.com/knights-lab/BURST/files/7286311/example.tar.gz).

I installed burst via bioconda and these are the commands I'm using: 
```
burst_linux_DB15 -r Y.region.fa -d DNA 31 -o Y.region.fa.edx -a Y.region.fa.acx -i 1.00 -s 1
burst_linux_DB15 -t 1 -q kmer-3161.fa -a Y.region.fa.acx -r Y.region.fa.edx -m ALLPATHS -o kmer-3161.region.txt
```

burst finds 13 hits (edit: and just 1 hit if I run with `-s`) but from my analysis I would expect 76 hits (see the bed file) and I'm pretty sure they are real hits:
```
> bedtools getfasta -fi Y.region.fa -bed kmer-3161.region.bed | grep -v "^>" | sort | uniq -c
	76 CTTCTCTTCTCTTCTCTTCTCTTCTCTTCTC
```
And this is how the hits are distributed in the region:
![region](https://user-images.githubusercontent.com/22707509/136035390-72c0a968-dd43-4657-b25b-82b1ef32f655.png)

So, in conclusion I think my question can be summed up as: how does ALLPATHS treat overlapping hits? I would say it reports just ~1 hit per overlapping window, but I'm not sure about that. Can you please help me out?

Best,
Luca

**Comment by GabeAl** [comment]
Hi Luca,

Happy to help. There may be a number of issues at play here, and in this case the first, which you've identified, seems the most plausible: 
- BURST indeed introduces a "deadzone" around each alignment for a query such that not more than 1 or 2 are reported in each alignment window. That is, a single query will find all alignments, with the caveat that only 1 valid alignment is reported per window. This is because many alignments with degenerate and near-identical solutions are possible within a small window and for most metagenomic analyses this is highly undesirable (the repeats we're expecting are regional like gene repeats, 16S copies, or reads that can hit multiple related genomes).
- Alignment window is defined by the -d and -s flags -- to make the window as small as possible, make the querylen is as low as possible and set -s to 1 [this is what you probably did above!]
- ALLPATHS is not ideal for this -- use FORAGE just in case there are overlaps that differ slightly in score. ALLPATHS only reports the very highest-scored ties. You'd probably want everything above a certain threshold.
- Try setting -i to .99 just in case there are some issues with alignment id

**Comment by ldenti** [comment]
Hi, thanks for the explanation. I'll try what you suggested.

I'm closing this: in case I need additional info, I'll reopen it.

Best,
Luca
